src/metrics.ts

```diff
@@ -218,6 +225,25 @@ export const MetricsProcessor = (
     eventBus.emit(MetricEvent.READY);
   };
 
+  const valueToString = (value: any): string => {
+    switch (typeof value) {
+      case 'string':
+        return value;
+      case 'number':
+      case 'boolean':
+      case 'bigint':
+      case 'symbol':
+        return value.toString();
+      case 'object':
+        if (Array.isArray(value)) {
+          return value.toString()
```

Comment:
For arrays of mixed types, e.g. ["hi", true] this just ensures that the format is in one enclosed string, which is why I didn't use JSON.Stringify.